### PR TITLE
Craftable Locker

### PR DESCRIPTION
**Pull request by iacore**
## About the PR

Currently this PR is a proof of concept. Here's a list of things that should be done before merge.

- [ ] adjust crafting material use
- make X craftable
  - [x] normal locker
  - [ ] secure locker (What's the cost of it?) (Is there a sprite for it already?)
  - [ ] wooden locker (wardrobe)
  - [ ] cabinet
- [ ] make door painter be able to paint locker into different colors
- [ ] sprites for generic lockers
- [ ] reorganize locker directory (don't put every locker sprite in one RSI)

## My design

- craft and deconstruct wooden closets (not in the game yet)
- craft and deconstruct wooden lockers (material use: 4 wood, 2 LV wires)
- craft and deconstruct steel closets (material use: 4 steel)
- craft and deconstruct steel lockers (material use: 4 steel, 2 LV wires)
- craft and deconstruct secure lockers (material use: 2 steel, 2 plasteel, 2 LV wires)

## Problem with existing code

currently, there are many closets:

- normal standing closets
- wall closets
- suit storage

they are many lockers as well (they share LockerBase)
- wood locker
- steel locker
- secure steel locker (harder to break)

I think this hierarchy needs to be cleaned up first, before this feature can be added.

For example, if I make LockerBase as deconstructable, then a visually wooden locker will give steel when broken down.

## Why / Balance
Everyone should be able to make lockers.

## Technical details
I have no idea how SS14 work.

## Media

https://github.com/space-wizards/space-station-14/assets/74560659/c2150c80-a2fe-4bdd-b45d-5fe31159541d

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
- add: Lockers can now be crafted!

### CODE REVIEW

**Comment by iacore** [comment]
I don't know why Locker doesn't reuse the Crate code. They are crates with a different look.

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/base_structurelockers.yml [hunk 1]
```diff
@@ -41,6 +41,16 @@
           SheetSteel1:
             min: 1
             max: 2
+  - type: Construction
+    graph: LockerGeneric
+    node: done
+    containers:
+    - entity_storage
+  - type: Reflect
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/base_structurelockers.yml:49]: why is this in a construction pr

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml [hunk 2]
```diff
@@ -1,3 +1,15 @@
+- type: entity
+  id: LockerGeneric
+  parent: LockerBase
+  name: a locker
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml:4]: no a

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/base_structurelockers.yml (hunk 1 above)
Comment (iacore) [reply Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/base_structurelockers.yml:49]: it is copied from the crate code. Might as well add it

File: Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+- type: constructionGraph
+  id: LockerGeneric
+  start: start
+  graph:
+    - node: start
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml:5]: style is to not indent - an extra level
```
graph:
- node: start
  edges:
  - to: done
    ...
```
etc

File: Resources/Textures/Structures/Storage/closet.rsi/meta.json [hunk 4]
```diff
@@ -347,6 +347,7 @@
     {
       "name": "hydro_open"
     },
+    { "name": "icon" },
```
Comment (deltanedas) [inline Resources/Textures/Structures/Storage/closet.rsi/meta.json:350]: style

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by lzk228** [comment]
maybe also add a closet craft?

File: Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml (hunk 3 above)
Comment (iacore) [reply Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml:5]: I literally copied this code!

Comment (deltanedas) [reply Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml:5]: yes and you have to update legacy code if you copy it

File: Resources/Textures/Structures/Storage/closet.rsi/meta.json (hunk 4 above)
Comment (deltanedas) [reply Resources/Textures/Structures/Storage/closet.rsi/meta.json:350]: its the style of the entire games rsis

**Comment by iacore** [comment]
> maybe also add a closet craft?

task added

File: Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml (hunk 3 above)
Comment (deltanedas) [reply Resources/Prototypes/Recipes/Crafting/Graphs/storage/lockergeneric.yml]: pro resolve

File: Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml (hunk 2 above)
Comment (deltanedas) [reply Resources/Prototypes/Entities/Structures/Storage/Closets/Lockers/lockers.yml]: pro resolve

**Comment by iacore** [comment]
I hate this

out

**Pull request closed without merging**